res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,412 @@
+    Prime4.rgbCode[deckColors[0] + "Dark"],
+    Prime4.rgbCode[deckColors[1] + "Dark"],
+    Prime4.rgbCode[deckColors[2] + "Dark"],
+    Prime4.rgbCode[deckColors[3] + "Dark"],
+];
+
+// Register '0x9n' as a button press and 'ox8n' as a button release
+components.Button.prototype.isPress = function(channel, control, value, status) {
+    return (status & 0xF0) === 0x90;
+};
+
+Prime4.init = function(_id, _debug) {
+    // Turn off all LEDs
+    midi.sendShortMsg(0x90, 0x75, 0x00);
+
+    // Initialise all three physical sections of the unit
+    Prime4.leftDeck = new Prime4.Deck([1, 3], 4);
+    Prime4.rightDeck = new Prime4.Deck([2, 4], 5);
+    Prime4.mixerA = new mixerStrip(1, 1);
+    Prime4.mixerB = new mixerStrip(2, 2);
+    Prime4.mixerC = new mixerStrip(3, 3);
+    Prime4.mixerD = new mixerStrip(4, 4);
+
+    // Initial LED values to set (Hopefully these will automatically initialize, but for now they won't.)
+    midi.sendShortMsg(0x90, 0x0D, colDeckDark[0]);                    // PFL 1
+    midi.sendShortMsg(0x91, 0x0D, colDeckDark[1]);                    // PFL 2
+    midi.sendShortMsg(0x92, 0x0D, colDeckDark[2]);                    // PFL 3
+    midi.sendShortMsg(0x93, 0x0D, colDeckDark[3]);                    // PFL 4
+    midi.sendShortMsg(0x9F, 0x1C, colDeck[0]);                        // Deck 1 Toggle
+    midi.sendShortMsg(0x9F, 0x1D, colDeck[1]);                        // Deck 2 Toggle
+    midi.sendShortMsg(0x9F, 0x1E, colDeckDark[2]);                    // Deck 3 Toggle
+    midi.sendShortMsg(0x9F, 0x1F, colDeckDark[3]);                    // Deck 4 Toggle
+    midi.sendShortMsg(0x94, 0x21, colDeck[0]);                        // Left Jog Wheel
+    midi.sendShortMsg(0x95, 0x21, colDeck[1]);                        // Right Jog Wheel
+    midi.sendShortMsg(0x94, 0x0B, Prime4.rgbCode.orange);             // Hot-Cue Button - Left Deck
+    midi.sendShortMsg(0x95, 0x0B, Prime4.rgbCode.orange);             // Hot-Cue Button - Right Deck
+    midi.sendShortMsg(0x9F, 0x0B, 1);                                 // Headphone Split Button
+    midi.sendShortMsg(0x9F, 0x0C, 2);                                 // Filter Sweep FX Button
+    midi.sendShortMsg(0x94, 0x1C, 1);                                 // Left Shift Button
+    midi.sendShortMsg(0x95, 0x1C, 1);                                 // Right Shift Button
+};
+
+Prime4.shutdown = function() {
+    // Return all LEDs to initial dim state
+    midi.sendShortMsg(0x90, 0x75, 0x01);
+};
+
+// All components contained in each mixer strip
+const mixerStrip = function(deckNumber, midiChannel) {
+    components.Deck.call(this, deckNumber);
+
+    // PFL Button
+    this.headphoneCue = new components.Button({
+        midi: [0x8F + midiChannel, 0x0D],
+        group: "[Channel" + midiChannel + "]",
+        key: "pfl",
+        on: colDeck[midiChannel - 1],
+        off: colDeckDark[midiChannel - 1],
+        type: components.Button.prototype.types.toggle,
+    });
```

Comment:
This should present you with an error if the `trigger` method was called. It should normally be called automatically and take care of the LED initialization for you.

```suggestion
    this.headphoneCue = new components.Button({
        midi: [0x8F + midiChannel, 0x0D],
        group: "[Channel" + midiChannel + "]",
        key: "pfl",
        on: colDeck[midiChannel - 1],
        off: colDeckDark[midiChannel - 1],
        type: components.Button.prototype.types.toggle,
        trigger: function() {
            throw "Success: trigger method called!";
        }
    });
```

Reply:
Looks like it is already :) I didn't need the manual MIDI messages in there after all

Reply:
Great.